### PR TITLE
refactor: use lerna to manage monorepo

### PR DESCRIPTION
**Pull request by neodmy**
Use Lerna to manage packages in this monorepo

## Description
Add Lerna to manage package releases. For the moment, I propose to manage the versions manually.

## Related Issue
#63 

## Motivation and Context
We want to release our 2 current packages in a more convenient way, meaning releasing new versions more automatically. There are options to automatize this, but the previous step should be using a tool that manages monorepos. I've thought of Lerna, as it's got some commands to do that. Next steps might be to incorporate a tool to analyze the commit message syntax to detect the right semver automatically. But for the moment I suggest we manage the versions ourselves.

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: packages/@guidesmiths/cuckoojs-cli/package.json [hunk 1]
```diff
@@ -48,13 +48,14 @@
...
-    "test": "npm run build && jest src/**/*.spec.ts",
+    "test": "jest src/**/*.spec.ts",
+    "test:dev": "npm run build && jest src/**/*.spec.ts",
     "lint": "eslint --ext .ts .",
     "lint:fix": "eslint --fix --ext .ts ."
   },
   "dependencies": {
     "@angular-devkit/schematics-cli": "^14.2.3",
-    "@guidesmiths/cuckoojs-schematics": "^0.0.2",
+    "@guidesmiths/cuckoojs-schematics": "*",
```
Comment (inigomarquinez) [inline packages/@guidesmiths/cuckoojs-cli/package.json:58]: I guess that with this lerna automatically knows the dependency tree and first publishes the `cuckoojs-schematics` package and then the `cuckoojs-cli` package, Is that right?

**Review by inigomarquinez — COMMENTED** [review]
LGTM!

File: packages/@guidesmiths/cuckoojs-cli/package.json (hunk 1 above)
Comment (neodmy) [reply packages/@guidesmiths/cuckoojs-cli/package.json:58]: Correct

Comment (neodmy) [reply packages/@guidesmiths/cuckoojs-cli/package.json:58]: To be more precise, I think Lerna handles the dependencies when doing the build.

**Review by jonatan-martinez — APPROVED** [review]

**Pull request merged**